blockchain/blockhashHandler.go

```diff
@@ -0,0 +1,128 @@
+package blockchain
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/atomic"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-sdk-go/data"
+)
+
+const (
+	minPollingInterval  = time.Second
+	minBlockTtl         = time.Second
+	logPath             = "blockchain/blockhashHandler"
+	blockByHashEndpoint = "blocks/%s?fields=timestamp"
+)
+
+type argsBlockhashHandler struct {
+	pollingInterval   time.Duration
+	blockTtl          time.Duration
+	httpClientWrapper httpClientWrapper
+}
+
+type blockhashHandler struct {
+	httpClientWrapper
+	log             logger.Logger
+	blockhashes     map[string]int
+	loopStatus      *atomic.Flag
+	cancel          func()
```

Comment:
this is never called.. the component should have a `Close() error` method which should call  `bh.cancel()`